### PR TITLE
Add some macros to make defining variables easier

### PR DESCRIPTION
**Pull request by YingboMa**
One can do stuff like
```julia
julia> @diffs D'''~t E'~t
((IndependentVariable(t),3), (IndependentVariable(t),1))

julia> @cons c=1 d=2
(Constant(1), Constant(2))

julia> @dvars x y z
(DependentVariable(x), DependentVariable(y), DependentVariable(z))

julia> @idvars t g
(IndependentVariable(t), IndependentVariable(g))

julia> @paras p b
(Parameter(p), Parameter(b))
```

### CODE REVIEW

**Comment by ChrisRackauckas** [comment]
Just some bikeshedding: `@params` and `@consts`. I don't think `@consts` is necessary since the methods should up-promote constants automatically, but there's no reason to not have it.

File: src/utils.jl [hunk 1]
```diff
@@ -1,5 +1,73 @@
+export @dvars, @idvars, @params, @diffs, @consts
+
 function expr_arr_to_block(exprs)
   block = :(begin end)
   foreach(expr -> push!(block.args, expr), exprs)
   block
 end
+
+# Build variables more easily
```
Comment (ChrisRackauckas) [inline src/utils.jl]: Move these to variables.jl

File: src/utils.jl [hunk 2]
```diff
@@ -1,5 +1,73 @@
...
+    end
+    push!(ex.args, Expr(:tuple, lhss...))
+    ex
+end
+
+macro consts(x...)
+    esc(_const_assign(x))
+end
+
+function count_order(x)
```
Comment (ChrisRackauckas) [inline src/utils.jl]: move these to operators.jl

**Comment by ChrisRackauckas** [comment]
We might as well make sure that the standard declarations can handle values like the comments: https://github.com/JuliaDiffEq/SciCompDSL.jl/issues/25#issuecomment-369975793 .

**Comment by YingboMa** [comment]
Is it better rename the macros to `@DV`, `@IV`, `@Param`, `@Const` and `@D`?

**Comment by AleMorales** [comment]
Not a big deal, but I would expand `@DV` and `@IV` to `@DVar` and `@IVar` and `@D` to `@Deriv`. Unless we want to really minimize typing, otherwise is confusing if some are very short and others not. And `@D` and `@DV` would be confusing for newcomers.

**Comment by ChrisRackauckas** [comment]
I agree with @AleMorales. The names were what I was using when I proposed the idea in https://github.com/JuliaDiffEq/DifferentialEquations.jl/issues/261 , but the structure was much more thought out than the names. I think you improved them, and then @AleMorales improved it one step more.

**Comment by AleMorales** [comment]
`@consts` can already be assigned values and expressions, you could use the same logic for the other inputs to implement #25

File: test/runtests.jl [hunk 3]
```diff
@@ -1,6 +1,32 @@
 using SciCompDSL
 using Base.Test
 
+@testset "Parsing Test" begin
+    @DVar x y=1 z
+    x1 = DependentVariable(:x)
+    y1 = DependentVariable(:y, 1)
+    z1 = DependentVariable(:z)
+    @test x1 === x
```
Comment (ChrisRackauckas) [inline test/runtests.jl]: use `isequal` for variable equality.

File: test/runtests.jl [hunk 4]
```diff
@@ -1,6 +1,32 @@
 using SciCompDSL
 using Base.Test
 
+@testset "Parsing Test" begin
+    @DVar x y=1 z
```
Comment (AleMorales) [inline test/runtests.jl]: Would be nice to use an expression like `z = 1 + div(5,9)` too and not only values in the tests

Comment (AleMorales) [reply test/runtests.jl]: Or something more sensible than what I wrote there,...

**Pull request merged**